addons/mail/static/src/core/common/composer.js

```diff
@@ -749,5 +754,6 @@ export class Composer extends Component {
         if (fullComposerContent) {
             composer.text = fullComposerContent;
         }
+        this.props.composer.emailAddSignature = !fullComposerContent;
```

Comment:
There is a `beforeunload` listener calling `saveContent`. When reloading the page, the system can save the composer text into the cache so that it will be restored when re-opening the composer. Because of this new condition, the system will disable the signature in that flow which is incorrect.

For recall, the signature should only be disabled when the message is restored from the full mail composer because, when opening the full-composer, we already add one.

I will hence save in the cache a boolean indicating whether the signature should be added or not.